components/customForm.vue

```diff
@@ -1,397 +1,424 @@
+      company_name: null,
+      contact_name: null,
+      phone: null,
+      email: null,
+      country: null
+    }
+  }),
+  mounted() {
+    this.formDefinition =
+      this.form === "volunteer" ? formVolunteer : formProject;
+    this.stepsNumber = this.formDefinition.length;
+    this.addstepProps();
+  },
+  methods: {
+    isTextInput() {
+      return this.stepProps.input === "text";
+    },
+    isNumberInput() {
+      return this.stepProps.input === "number";
+    },
+    isEmailInput() {
+      return this.stepProps.input === "email";
+    },
+    isTelInput() {
+      return this.stepProps.input === "tel";
+    },
+    isNoneInput() {
+      return this.stepProps.input === "none";
+    },
+    isOptionInput() {
+      return this.stepProps.input === "option";
+    },
+    shouldDisplayAcceptButton() {
+      return (
+        this.isTextInput() ||
+        this.isNumberInput() ||
+        this.isEmailInput() ||
+        this.isTelInput() ||
+        this.isOptionInput()
+      );
+    },
+    getInputOptions() {
+      if (this.stepProps.input !== "button") {
+        return [];
+      }
+
+      return this.stepProps.options;
+    },
+    goToStep(step) {
+      this.submittedField = false;
+
+      if (step === -1) {
+        this.displayCaptcha = true;
+        return;
+      }
+
+      this.step = step;
+      this.addstepProps();
+    },
+    findStepById(stepId) {
+      for (let i = 0; i < this.formDefinition.length; i++) {
+        if (this.formDefinition[i].id === stepId) {
+          return this.formDefinition[i];
+        }
+      }
+
+      return null;
+    },
+    addstepProps() {
+      for (let i = 0; i < this.formDefinition.length; i++) {
+        if (this.formDefinition[i].step === this.step) {
+          if (this.formDefinition[i].condition) {
+            const condition = this.formDefinition[i].condition;
+            if (
+              !condition.type &&
+              this.formData[condition.id] !== condition.value
+            ) {
+              this.goToStep(this.formDefinition[i].next);
+              break;
+            } else if (
+              condition.type === "different" &&
+              this.formData[condition.id] === condition.value
+            ) {
+              this.goToStep(this.formDefinition[i].next);
+              break;
+            }
+          }
+          this.stepProps = this.formDefinition[i];
+        }
+      }
+    },
+    storeOptionInfo(option) {
+      this.formData[this.stepProps.id] = option.text;
+      this.goToStep(option.next);
+    },
+    storeStepInfo() {
+      this.submittedField = true;
+      const isValid = this.$refs[this.stepProps.id].checkValidity();
+      this.$refs[this.stepProps.id].reportValidity();
+
+      if (isValid) {
+        this.formData[this.stepProps.id] = this.$refs[
+          this.stepProps.id
+        ].localValue;
+        this.goToStep(this.stepProps.next);
+      }
+    },
+    submitForm() {
+      const data = {

```

Comment:
this field mapping will need to be adjusted to the api to push new volunteers/projects.  I didn't find an implementation of it so I just made up some ids on my own